docs/src/whatsnew/latest.rst

```diff
@@ -48,8 +48,8 @@ This document explains the changes made to Iris for this release
 🚀 Performance Enhancements
 ===========================
 
-#. N/A
-
+#. `@bouweandela`_ made :meth:`iris.cube.CubeList.concatenate` faster if more
+   than two cubes are concatenated. (:pull:`5926`)
```

Comment:
I discovered that the function `benchmarks.generate_data.stock.realistic_4d_w_everything` seems to produce lazy data for the variable data as well as the ancillary variable data regardless of whether it is called with `lazy=True` or `lazy=False`. Maybe that is something to look into to improve the memory benchmark? In any case, it leads to the following:

The increase in memory use comes from computing the hashes for the lazy `"quality_flags"` ancillary variable in the benchmark cube. In the current `main` branch, this ancillary variable is never accessed (so not loaded into memory) because it spans the concatenation axis and will therefore be concatenated so there is no need to compare if it is equal in the input cubes. I do not see a way to avoid this (the hashes are needed to compute the concatenation axis) and the performance impact seems minimal since the implementation here is considerably faster even with this effect. I added a note to the whatsnew entry in 1969d49.